airflow/providers/google/common/hooks/base_google.py

```diff
@@ -639,6 +689,7 @@ async def get_sync_hook(self) -> Any:
             self._sync_hook = await sync_to_async(self.sync_hook_class)(**self._hook_kwargs)
         return self._sync_hook
 
-    async def service_file_as_context(self) -> Any:
```

Comment:
You are right! I only checked the use of this method within the airflow repo, but indeed it's public.

I can keep it in. Doesn't necessarily need to be deprecated though.

In this context I will also make CredentialsToken non-public, to avoid issues in the future.

Reply:
@pankajastro I've added some changes to address you comment.